src/support/rand.rs

```diff
@@ -7,20 +7,61 @@
 
 use crate::Uint;
 use rand::{
-    distributions::{Distribution, Standard, Uniform},
+    distributions::{Distribution, Standard},
     Rng,
 };
 
 impl<const BITS: usize, const LIMBS: usize> Distribution<Uint<BITS, LIMBS>> for Standard {
+    #[inline]
     fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Uint<BITS, LIMBS> {
-        let mut limbs = [0; LIMBS];
-        if let Some((last, rest)) = limbs.split_last_mut() {
-            for limb in rest {
-                *limb = rng.gen();
-            }
-            *last = Uniform::new_inclusive(0, Uint::<BITS, LIMBS>::MASK).sample(rng);
-        }
-        Uint::<BITS, LIMBS>::from_limbs(limbs)
+        <Uint<BITS, LIMBS>>::random_with(rng)
+    }
+}
+
+impl<const BITS: usize, const LIMBS: usize> Uint<BITS, LIMBS> {
+    /// Creates a new [`Uint`] with the default cryptographic random number
+    /// generator.
+    ///
+    /// This is currently [`rand::thread_rng`].
+    #[inline]
+    #[must_use]
+    #[cfg(feature = "std")]
+    pub fn random() -> Self {
+        // SAFETY: `uint` is only accessible after random initialization.
+        #[allow(clippy::uninit_assumed_init)]
+        let mut uint = unsafe { core::mem::MaybeUninit::<Self>::uninit().assume_init() };
```

Comment:
why unsafety here instead of `Uint::ZERO`?

Reply:
uses undefined memory instead of zeroing it